src/nvim/api/private/dispatch.c

```diff
@@ -0,0 +1,47 @@
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <assert.h>
+#include <msgpack.h>
+
+#include "nvim/map.h"
+#include "nvim/log.h"
+#include "nvim/vim.h"
+#include "nvim/msgpack_rpc/helpers.h"
+#include "nvim/api/private/dispatch.h"
+#include "nvim/api/private/helpers.h"
+#include "nvim/api/private/defs.h"
+
+#include "nvim/api/buffer.h"
+#include "nvim/api/tabpage.h"
+#include "nvim/api/ui.h"
+#include "nvim/api/vim.h"
+#include "nvim/api/window.h"
+
+static Map(String, MsgpackRpcRequestHandler) *methods = NULL;
+
+static void msgpack_rpc_add_method_handler(String method, MsgpackRpcRequestHandler handler)
+{
+  map_put(String, MsgpackRpcRequestHandler)(methods, method, handler);
+}
+
+MsgpackRpcRequestHandler msgpack_rpc_get_handler_for(const char *name,
+                                                     size_t name_len)
+{
+  String m = {
+    .data=(char *)name,
+    .size=name_len, //TODO(bfredl): was the max_fname_len check important?
```

Comment:
Aren't optimizations supposed to be done for code paths that are likely to happen? While invalid method calls could happen, a client calling nvim with a superlong invalid method name sounds like an incredibly rare event, and would already have wasted cpu cycles on the client side.... Anyway either gperf or triehash will handle this for us when we switch to it here.